forch/forchestrator.py

```diff
@@ -436,9 +445,43 @@ def _restore_states(self):
     def _restore_faucet_config(self, timestamp, config_hash):
         config_info, faucet_dps, _ = self._get_faucet_config()
         self._update_config_warning_varz()
-        assert config_hash == config_info['hashes'], 'config hash info does not match'
+
+        if config_hash == config_info['hashes']:
+            self._attempt_cancel_config_hash_clash_timer()
+        else:
+            LOGGER.warning('Config hash does not match')
+            self._attempt_start_config_hash_clash_timer()
+
         self._faucet_collector.process_dataplane_config_change(timestamp, faucet_dps)
 
+    def _attempt_start_config_hash_clash_timer(self):
+        if self._config_hash_clash_timer and self._config_hash_clash_timer.is_alive():
+            return
+        self._config_hash_clash_timer = threading.Timer(
+            interval=self._config_hash_clash_timeout_sec,
+            function=self._check_config_hash_clashed)
+        self._config_hash_clash_timer.start()
+        LOGGER.info(
+            'Config hash clash timer started with %s seconds', self._config_hash_clash_timeout_sec)
+
+    def _attempt_cancel_config_hash_clash_timer(self):
+        if not self._config_hash_clash_timer or not self._config_hash_clash_timer.is_alive():
+            return
+        self._config_hash_clash_timer.cancel()
+        LOGGER.info('Config hash clash timer cancelled')
+
+    def _check_config_hash_clashed(self):
+        _, varz_config_hashes = self._get_varz_config()
+        config_info, _, _ = self._get_faucet_config()
+
+        if varz_config_hashes != config_info['hashes']:
+            LOGGER.error(
+                'Config hash does not match after %s seconds', self._config_hash_clash_timeout_sec)
+            self._config_hash_clashed = True
+
+    def _get_config_hash_clashed(self):
```

Comment:
private getter is redundant -- just use the variable directly

Reply:
removed